Obsidian/WorldData/World.cs

```diff
@@ -529,11 +529,13 @@ public async Task ManageChunksAsync()
 
         public async Task FlushRegionsAsync()
         {
-            await Task.Run(() =>
-            {
-                Server.BroadcastMessage("Saving to disk...");
-                Parallel.ForEach(Regions.Values, async r => await r.FlushAsync());
-                Server.BroadcastMessage("Save complete.");
+            await Task.WhenAll(new List<Task> {
+                Task.Run(() =>
+                {
+                    Server.BroadcastMessage("Saving to disk...");
+                    Parallel.ForEach(Regions.Values, async r => await r.FlushAsync());
+                    Server.BroadcastMessage("Save complete.");
+                })
```

Comment:
You could just do `await Task.WhenAll(Regions.Values.Select(r => r.FlushAsync()))` and move the broadcast message back outside ^ ^